snapcraft/internal/pluginhandler/__init__.py

```diff
@@ -548,33 +567,16 @@ def prepare_build(self, force=False):
     def build(self, force=False):
         self.makedirs()
 
-        if not self.plugin.out_of_source_build:
-            if os.path.exists(self.plugin.build_basedir):
-                shutil.rmtree(self.plugin.build_basedir)
-
-            # FIXME: It's not necessary to ignore here anymore since it's now
-            # done in the Local source. However, it's left here so that it
-            # continues to work on old snapcraft trees that still have src
-            # symlinks.
-            def ignore(directory, files):
-                if directory == self.plugin.sourcedir:
-                    snaps = glob(os.path.join(directory, "*.snap"))
-                    if snaps:
-                        snaps = [os.path.basename(s) for s in snaps]
-                        return common.SNAPCRAFT_FILES + snaps
-                    else:
-                        return common.SNAPCRAFT_FILES
-                else:
-                    return []
+        if not (
+            isinstance(self.plugin, plugins.v1.PluginV1)
+            and self.plugin.out_of_source_build
+        ):
+            if os.path.exists(self.part_build_dir):
```

Comment:
so this will always happen with v2?

Reply:
I suppose this is what we just discussed not doing and you were going to look into. I can remove the rm entirely and we can see what happens

Reply:
i thought we discussed the opposite.  v2 would leave the build directory alone and incremental builds would work?

Reply:
exactly, but I did this before that conversation, that's all I am saying :-)

Reply:
I say let's make the change now :)